src/core/graph.ts

```diff
@@ -150,17 +150,11 @@ export class DependencyGraph<T> {
     }
   }
 
-  getEdges(node: T): Set<T> {
-    // FIXME: use different terminology: origins/effects
-    const nodes = this.adjacencyMap.get(node);
-    if (nodes !== undefined) {
-      return nodes;
-    } else {
-      return new Set<T>();
-    }
+  getOriginsOfEffect(node: T): Set<T> {
+    return this.adjacencyMap.get(node) ?? new Set<T>();
   }
 
-  getBackEdges(node: T): Set<T> {
+  getEffectsOfOrigin(node: T): Set<T> {
```

Comment:
Using `origin` and `effect` very effectively eliminated the confusion while using the API; yet it is a bit confusing when writing functions within the class that are dependent upon these functions

Reply:
Maybe using upstream/downstream would be even better? E.g., `getUpstreamNeighbors` or `getUpstreamNodes` or simply `getUpstream`.

Reply:
We could also use source/sink terminology...